File: tasks/day1/erlang/main.erl
-module(main).
-export([start/1]).

start([InputFile]) ->
    {ok, Data} = file:read_file(InputFile),
    {Level, Basemant, _} = lists:foldl(fun(Sym, {Level, Basemant, Index}) ->
            NewLevel = case Sym of
                $( -> Level + 1;
                $) -> Level - 1;
                _ -> Level
            end,
            NewBasemant = case NewLevel of
                -1 when Basemant == 0 -> Index;
                _ -> Basemant
            end,
            {NewLevel, NewBasemant, Index + 1}
        end, {0, 0, 1}, binary_to_list(Data)),
    io:format("Level: ~w~nBasemant: ~w~n", [Level, Basemant]).

File: tasks/day1/golang/main.go
package main

import (
    "fmt"
    "os"
    "io/ioutil"
)

func main() {
    var level, basemant_level int
    file_path := os.Args[1]
    content, _ := ioutil.ReadFile(file_path)

    for index, elm := range content {
        if elm == '(' {
            level++
        } else if elm == ')' {
            level--
        }

        if level < 0 && basemant_level == 0 {
            basemant_level = index + 1
        }
    }

    fmt.Printf("Level: %d\nBasemant: %d\n", level, basemant_level)
}